### PR TITLE
add support  Android Studio Koala Feature Drop | 2024.1.2  (update libs local version)

### PR DESCRIPTION
**Pull request by CMingTseng**
Dear Sir

1.  modify untilBuild  -- like  152cd48c8135cb591df42bb44ed079871d593e1f

2.  update local lib version :  ProjectVersions &  versions.json <-- some suguest 

__bug & issue ___

1. use this plugin create project lost "gradle/wrapper/gradle-wrapper.properties" --> at kotlin 1.9.10 will let iOS build fail  / over kotlin 2.0.0 will let "project init fail" .  can fix it ?

2. when kotlin 2.0.0 the compose compiler(Ref : https://developer.android.com/jetpack/androidx/releases/compose-compiler & https://www.jetbrains.com/help/kotlin-multiplatform-dev/compose-compiler.html#migrating-a-jetpack-compose- ) merge to kotlin  maybe the UI can set kotlin version ? when over 2.0.0 will add compose compiler pluigin ?
and enable "composeResources" !! ( the project template can use default icon for AD for you )

3.  the ktor /koin version use "https://raw.githubusercontent.com/jakepurple13/Full-Multiplatform-Compose-Plugin/main/versions.json" is hard code !!  can modify to real check ? ( ref : https://github.com/gradle-plugins/portal-api ) & the UI  can like set AppName or packagename ?  when check "use remote source" call api get ?

4. the DI (dependency injection) can let switch ?  koin & hilt ?

5. can like Android Template(Ref : https://github.com/cnrture/QuickProjectWizard/tree/main/src/main/kotlin/com/github/cnrture/quickprojectwizard/arch. 
 & https://developer.android.com/studio/projects/templates) create "MVVM" Template ?

### CODE REVIEW

File: src/main/kotlin/com/programmersbox/fullmultiplatformcompose/BuilderParams.kt [hunk 1]
```diff
@@ -10,7 +10,7 @@ class BuilderParams {
     var hasiOS: Boolean by mutableStateOf(false)
     var hasDesktop: Boolean by mutableStateOf(false)
 
-    var packageName by mutableStateOf("com.example")
+    var packageName by mutableStateOf("idv.neo.compose")
```
Comment (jakepurple13) [inline src/main/kotlin/com/programmersbox/fullmultiplatformcompose/BuilderParams.kt:13]: Can you please change this back to `com.example`?

File: src/main/kotlin/com/programmersbox/fullmultiplatformcompose/BuilderParams.kt [hunk 2]
```diff
@@ -28,7 +28,7 @@ class Android {
 }
 
 class IOS {
-    var appName: String by mutableStateOf("My Application")
+    var appName: String by mutableStateOf("CMP_With_KMP App")
```
Comment (jakepurple13) [inline src/main/kotlin/com/programmersbox/fullmultiplatformcompose/BuilderParams.kt:31]: I don't think there needs to be any underlines here.

File: src/main/resources/fileTemplates/code/project_settings.gradle.kts.ft [hunk 3]
```diff
@@ -1,12 +1,48 @@
...
+                includeGroupAndSubgroups("com.android")
+                includeGroupAndSubgroups("com.google")
+            }
+        }
         mavenCentral()
         maven("https://maven.pkg.jetbrains.space/public/p/compose/dev")
+        maven(url = "https://jitpack.io")
+        maven(url = "https://s3.amazonaws.com/repo.commonsware.com")
+        maven(url = "https://jogamp.org/deployment/maven")
+    }
```
Comment (jakepurple13) [inline src/main/resources/fileTemplates/code/project_settings.gradle.kts.ft:23]: What are these for?

File: src/main/resources/fileTemplates/code/project_settings.gradle.kts.ft [hunk 4]
```diff
@@ -1,12 +1,48 @@
...
+                includeGroupAndSubgroups("androidx")
+                includeGroupAndSubgroups("com.android")
+                includeGroupAndSubgroups("com.google")
+            }
+        }
+        mavenCentral()
+        maven(url = "https://jitpack.io")
+        maven(url = "https://s3.amazonaws.com/repo.commonsware.com")
+        maven(url = "https://maven.pkg.jetbrains.space/public/p/compose/dev")
+        maven(url = "https://jogamp.org/deployment/maven")
```
Comment (jakepurple13) [inline src/main/resources/fileTemplates/code/project_settings.gradle.kts.ft:43]: Here too minus the middle one.

File: versions.json [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
+  "kotlinVersion": "2.0.20",
+  "agpVersion": "8.5.0",
+  "composeVersion": "1.6.11",
+  "androidxAppCompat": "1.7.0",
+  "androidxCore": "1.13.1",
   "ktor": "2.3.6",
-  "koin": "3.4.0"
+  "koin": "3.4.0",
+  "ktorfit": "1.12.0",
+  "napier": "2.6.1"
```
Comment (jakepurple13) [inline versions.json:10]: What's napier again?

Comment (jakepurple13) [reply versions.json:10]: I also see this being used anywhere.

**Review by jakepurple13 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/programmersbox/fullmultiplatformcompose/BuilderParams.kt (hunk 2 above)
Comment (CMingTseng) [reply src/main/kotlin/com/programmersbox/fullmultiplatformcompose/BuilderParams.kt:31]: sorry

I sent the wrong PR

I mainly want to change the versions of various libs (especially the kotlin part, which I want to upgrade to 2.0.20 or above)

I also added my commonly used repos and libs, and then sent the PR without distinguishing them.

Mainly I want to know if this plugin can be modified to support kotlin 2.0 or above and more lib and mvvm templates?

File: src/main/kotlin/com/programmersbox/fullmultiplatformcompose/BuilderParams.kt (hunk 1 above)
Comment (CMingTseng) [reply src/main/kotlin/com/programmersbox/fullmultiplatformcompose/BuilderParams.kt:13]: sorry

I sent the wrong PR

I mainly want to change the versions of various libs (especially the kotlin part, which I want to upgrade to 2.0.20 or above)

I also added my commonly used repos and libs, and then sent the PR without distinguishing them.

Mainly I want to know if this plugin can be modified to support kotlin 2.0 or above and more lib and mvvm templates?

File: versions.json (hunk 5 above)
Comment (CMingTseng) [reply versions.json:10]: sorry

I sent the wrong PR

I mainly want to change the versions of various libs (especially the kotlin part, which I want to upgrade to 2.0.20 or above)

I also added my commonly used repos and libs, and then sent the PR without distinguishing them.

Mainly I want to know if this plugin can be modified to support kotlin 2.0 or above and more lib and mvvm templates?

File: src/main/resources/fileTemplates/code/project_settings.gradle.kts.ft (hunk 4 above)
Comment (CMingTseng) [reply src/main/resources/fileTemplates/code/project_settings.gradle.kts.ft:43]: sorry

I sent the wrong PR

I mainly want to change the versions of various libs (especially the kotlin part, which I want to upgrade to 2.0.20 or above)

I also added my commonly used repos and libs, and then sent the PR without distinguishing them.

Mainly I want to know if this plugin can be modified to support kotlin 2.0 or above and more lib and mvvm templates?

File: src/main/kotlin/com/programmersbox/fullmultiplatformcompose/BuilderParams.kt (hunk 1 above)
Comment (jakepurple13) [reply src/main/kotlin/com/programmersbox/fullmultiplatformcompose/BuilderParams.kt:13]: It definitely can. I just don't have the time to work on it plus there's the official web wizard which works well

**Pull request closed without merging**